### PR TITLE
[CLI-3095] On kafka topic update --config num.partitions=XXX, poll for updates or print warning

### PR DESCRIPTION
**Pull request by tmalikconfluent**
Release Notes
-------------
Bug Fixes
- CLI shows the correct partition count on `confluent kafka topic update my-topic --config num.partitions=X` command

Checklist
---------
- [x] Leave this box unchecked if features are not yet available in production

What
----
On `kafka topic update --config num.partitions=XXX`, poll for updates or print warning

References
----------
https://confluentinc.atlassian.net/browse/CLI-3095

Test & Review
-------------
Updated the integration tests.

### CODE REVIEW

File: internal/kafka/command_topic_update.go [hunk 1]
```diff
@@ -114,11 +116,39 @@ func (c *command) update(cmd *cobra.Command, args []string) error {
...
 			return err
 		}
-		updateResp, err := kafkaREST.CloudClient.UpdateKafkaTopicPartitionCount(topicName, kafkarestv3.UpdatePartitionCountRequestData{PartitionsCount: int32(updateNumPartitionsInt)})
+		_, err = kafkaREST.CloudClient.UpdateKafkaTopicPartitionCount(topicName, kafkarestv3.UpdatePartitionCountRequestData{PartitionsCount: int32(updateNumPartitionsInt)})
 		if err != nil {
 			return err
 		}
-		configsValues[numPartitionsKey] = fmt.Sprint(updateResp.PartitionsCount)
+
+		errPartition := retry.Retry(time.Second/10, time.Second, func() error {
```
Comment (brianstrauch) [inline internal/kafka/command_topic_update.go:124]:
```suggestion
		err := retry.Retry(100 * time.Millisecond, time.Second, func() error {
```

File: internal/kafka/command_topic_update.go [hunk 2]
```diff
@@ -114,11 +116,39 @@ func (c *command) update(cmd *cobra.Command, args []string) error {
...
+			topic, httpRespPartition, err := kafkaREST.CloudClient.GetKafkaTopic(topicName)
+			if err != nil {
+				if restErr, parseErr := kafkarest.ParseOpenAPIErrorCloud(err); parseErr == nil && restErr.Code == ccloudv2.UnknownTopicOrPartitionErrorCode {
+					return fmt.Errorf(errors.UnknownTopicErrorMsg, topicName)
+				}
+				return kafkarest.NewError(kafkaREST.CloudClient.GetUrl(), err, httpRespPartition)
+			}
+
+			if int64(topic.PartitionsCount) != (updateNumPartitionsInt) {
```
Comment (brianstrauch) [inline internal/kafka/command_topic_update.go:133]:
```suggestion
			if int64(topic.PartitionsCount) != updateNumPartitionsInt {
```

File: internal/kafka/command_topic_update.go [hunk 3]
```diff
@@ -114,11 +116,39 @@ func (c *command) update(cmd *cobra.Command, args []string) error {
...
+				return fmt.Errorf(`the partition count not yet updated`)
```
Comment (brianstrauch) [inline internal/kafka/command_topic_update.go:134]:
```suggestion
				return fmt.Errorf("the partition count is not yet updated")
```

File: internal/kafka/command_topic_update.go [hunk 4]
```diff
@@ -114,11 +116,39 @@ func (c *command) update(cmd *cobra.Command, args []string) error {
...
+			}
+			return nil
+		})
+
```
Comment (brianstrauch) [inline internal/kafka/command_topic_update.go:138]:
```suggestion
```

File: internal/kafka/command_topic_update.go [hunk 5]
```diff
@@ -114,11 +116,39 @@ func (c *command) update(cmd *cobra.Command, args []string) error {
...
+		if errPartition != nil {
+			return errPartition
+		}
+
+		topic, httpRespPartition, errUpdated := kafkaREST.CloudClient.GetKafkaTopic(topicName)
+		if errUpdated != nil {
```
Comment (brianstrauch) [inline internal/kafka/command_topic_update.go:144]:
```suggestion
		topic, httpResp, err := kafkaREST.CloudClient.GetKafkaTopic(topicName)
		if err != nil {
```

File: test/test-server/kafka_rest_router.go [hunk 6]
```diff
@@ -541,6 +541,12 @@ func handleKafkaRestTopic(t *testing.T) http.HandlerFunc {
 				TopicName:       topic,
 				PartitionsCount: 3,
 			}
+			if topic == "topic-exist-rest" {
+				data = cckafkarestv3.TopicData{
+					TopicName:       topic,
+					PartitionsCount: 6,
+				}
```
Comment (brianstrauch) [inline test/test-server/kafka_rest_router.go:548]:
```suggestion
				data.PartitionsCount = 6
```

**Review by brianstrauch — DISMISSED** [review]

**Comment by tmalikconfluent** [comment]
opened new PR

**Pull request closed without merging**